rotkehlchen/exchanges/coinbase.py

```diff
@@ -442,6 +488,57 @@ def query_online_trade_history(
             if trade and trade.timestamp >= start_ts and trade.timestamp <= end_ts:
                 trades.append(trade)
 
+        # Analyze convertions of coins. We address them as sells
+        raw_transactions = []
+        for account_id in account_ids:
+            raw_transactions.extend(self._api_query(f'accounts/{account_id}/transactions'))
+
+        # Maps every trade id to their two transactions
+        trade_pairs = defaultdict(list)
+        for transaction in raw_transactions:
+            if 'type' in transaction.keys() and transaction['type'] == 'trade':
+                trade_pairs[transaction['trade']['id']].append(transaction)
```

Comment:
What `KeyError` you are referring here? `type` in the transaction dict because I verify `'type' in transaction.keys()` and 
`transaction['trade']` has always the `id` field. Also `trade_pairs` is a `defaultdict` so if the key is missing it uses the default.

Reply:
> transaction['trade'] has always the id field. Also trade_pairs is a defaultdict so if the key is missing it uses the default.

According to whom? The docs? Doesn't matter. Docs can be wrong, or something else can go wrong. External input should never be trusted.